benchmarks/benchmarks.rb

```diff
@@ -1,41 +1,86 @@
 GITHUB_MAIN = "MemoWise_GitHubMain"
-
-# We download a the main branch of MemoWise on GitHub into a tmp directory to
-# compare against the local version when we run benchmarks
-Dir.mktmpdir do |directory|
-  Dir["#{github_memo_wise_path}/lib/**/*.rb"].each do |file|
-    Tempfile.open([File.basename(file)[0..-4], ".rb"], directory) do |tempfile|
-      tempfile.write(File.read(file).gsub("MemoWise", GITHUB_MAIN))
-      tempfile.rewind
-      require tempfile.path
-    end
-  end
-end
-
+GITHUB_MAIN_BENCHMARK_NAME = "memo_wise-git-main"
+LOCAL_BENCHMARK_NAME = "memo_wise-local"
+
+# 1. GitHub version of MemoWise and the local source of MemoWise, share a namespace
+# 2. memery & alt_memery share the namespace Memery
+# 3. memoist & memoist3 share the namespace Memoist, and also share a load path for their version.rb files.
+# This means we must `require: false` in `benchmarks/Gemfile` all, or all but one, of each of these duplicates,
+#   or we take care to only load them in discrete Ruby versions,
+#   to avoid a namespace collision before re-namespacing duplicates
+re_namespaced_gems = [
+  GemBench::Jersey.new(
+    gem_name: "memo_wise",
+    trades: {
+      "MemoWise" => GITHUB_MAIN
+    },
+    metadata: {
+      activation_code: "prepend #{GITHUB_MAIN}",
+      memoization_method: :memo_wise,
+    },
+  ),
+  GemBench::Jersey.new(
+    gem_name: "alt_memery",
+    trades: {
+      "Memery" => "AltMemery"
+    },
+    metadata: {
+      activation_code: "include AltMemery",
+      memoization_method: :memoize,
+    },
+  ),
+  GemBench::Jersey.new(
+    gem_name: "memoist3",
+    trades: {
+      "Memoist" => "MemoistThree"
+    },
+    metadata: {
+      activation_code: "extend MemoistThree",
+      memoization_method: :memoize,
+    },
+  ),
+  GemBench::Jersey.new(
+    gem_name: "memoist",
+    trades: {
+      "Memoist" => "MemoistOne"
+    },
+    metadata: {
+      activation_code: "extend MemoistOne",
+      memoization_method: :memoize,
+    },
+  ),
+].each(&:doff_and_don) # Copies, re-namespaces, and requires each gem.
+
+# Here we require memo_wise gem from the `pwd` source.
+# Separately, in the Gemfile, we download the main branch of MemoWise from GitHub.
+# We've already moved it into a tmp directory and re-namespaced it to allow benchmark comparison with local source.
 require_relative "../lib/memo_wise"
 
 # Some gems do not yet work in Ruby 3 so we only require them if they're loaded
-# in the Gemfile.
-%w[memery memoist memoized memoizer ddmemoize dry-core].
+# in the Gemfile.  Gems re-namespaced by GemBench::Jersey will have already been loaded by now.
+%w[memery memoized memoizer ddmemoize dry-core].
   each { |gem| require gem if Gem.loaded_specs.key?(gem) }
 
+# Some Gems Have Modules Which Need To Be Required Manually:
+# 1. `dry-core` => Memoizable
+{
+  # file_path => gem_name
+  "dry/core/memoizable" => "dry-core",
```

Comment:
This feels a bit like overkill to me if this is the only gem we have like this. Thoughts on changing it back to the old version?

Reply:
It had included other gems / files during the convulsions it went through to get where it is, and it is definitely overkill now.